batch/batch/worker/worker.py

```diff
@@ -1812,6 +2138,22 @@ def __init__(self, client_session: httpx.ClientSession):
         self.headers = None
         self.compute_client = None
 
+        self.jvm_initializer_task = asyncio.ensure_future(self._initialize_jvms())
+        self.jvms: List[JVM] = []
+
+    async def _initialize_jvms(self):
+        self.jvms = await asyncio.gather(*[JVM.create(i) for i in range(CORES)])
+        log.info(f'JVMs initialized {self.jvms}')
+
+    async def borrow_jvm(self) -> JVM:
+        await self.jvm_initializer_task
+        assert self.jvms
+        return self.jvms.pop()
```

Comment:
Hmm. I don't think a Queue makes sense because it's an error if a Job wants to borrow a JVM and the JVM does not exist. The design of this system is intended to ensure there are never more JVMJobs than JVMs on a machine.

Fair point about the global synchronization point. I guess I could make self.jvms a list of `Future[JVM]` and do a `gather(..., FIRST_COMPLETED)`. I'm open to this change if JVM startup time is variable and a bottleneck for Query on Batch.